backend/video/src/subtitle.py

```diff
@@ -35,85 +37,67 @@ def get_subtitles(self):
-        if not all_formats:
-            return False
+            return {}
 
-        subtitle_json3 = [i for i in all_formats if i["ext"] == "json3"]
-        if not subtitle_json3:
-            print(f"{self.video.youtube_id}-{lang}: json3 not processed")
-            return False
-
-        subtitle = subtitle_json3[0]
-        subtitle.update(
-            {"lang": lang, "source": "auto", "media_url": media_url}
-        )
-
-        return subtitle
-
-    def _normalize_lang(self):
-        """normalize country specific language keys"""
-        all_subtitles = self.video.youtube_meta.get("subtitles")
-        if not all_subtitles:
-            return False
-
-        all_keys = list(all_subtitles.keys())
-        for key in all_keys:
-            lang = key.split("-")[0]
-            old = all_subtitles.pop(key)
+        candidate_subtitles = {}
+        for lang, all_formats in all_subtitles.items():
             if lang == "live_chat":
+                # not supported yet
                 continue
-            all_subtitles[lang] = old
-
-        return all_subtitles
 
-    def _get_user_subtitles(self, lang):
-        """get subtitles uploaded from channel owner"""
-        print(f"{self.video.youtube_id}-{lang}: get user uploaded subtitles")
-        all_subtitles = self._normalize_lang()
-        if not all_subtitles:
-            return False
+            video_media_url = self.video.json_data["media_url"]
+            media_url = video_media_url.replace(".mp4", f".{lang}.vtt")
+            if not all_formats:
+                # no subtitles found
+                continue
 
-        video_media_url = self.video.json_data["media_url"]
-        media_url = video_media_url.replace(".mp4", f".{lang}.vtt")
-        all_formats = all_subtitles.get(lang)
-        if not all_formats:
-            # no user subtitles found
-            return False
+            subtitle_json3 = [i for i in all_formats if i["ext"] == "json3"]
+            if not subtitle_json3:
+                print(f"{self.video.youtube_id}-{lang}: json3 not processed")
+                continue
 
-        subtitle = [i for i in all_formats if i["ext"] == "json3"][0]
-        subtitle.update(
-            {"lang": lang, "source": "user", "media_url": media_url}
-        )
+            subtitle = subtitle_json3[0]
+            subtitle.update(
+                {"lang": lang, "source": source, "media_url": media_url}
+            )
+            candidate_subtitles[lang] = subtitle
 
-        return subtitle
+        return candidate_subtitles
 
     def download_subtitles(self, relevant_subtitles):
         """download subtitle files to archive"""
+        subtitle_list = ", ".join(map(itemgetter("lang"), relevant_subtitles))
+        print(
+            f"{self.video.youtube_id}: downloading subtitles: {subtitle_list}"
+        )
         videos_base = EnvironmentSettings.MEDIA_DIR
         indexed = []
         for subtitle in relevant_subtitles:
```

Comment:
I tried to use the "all" method here, I got blocked quite fast. Better to slow these requests down here. you can use the `rand_sleep` function from the common/src/helper module.